test/integration/advisory_locks_test.go

```diff
@@ -0,0 +1,70 @@
+package integration
+
+import (
+	"context"
+	"net/http"
+	"sync"
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"gitlab.cee.redhat.com/service/sdb-ocm-example-service/pkg/api/openapi"
+	"gitlab.cee.redhat.com/service/sdb-ocm-example-service/pkg/db"
+	"gitlab.cee.redhat.com/service/sdb-ocm-example-service/test"
+)
+
+func TestAdvisoryLocks_UpdateDinosaurWithRacingRequests(t *testing.T) {
+	h, client := test.RegisterIntegration(t)
+
+	account := h.NewRandAccount()
+	ctx := h.NewAuthenticatedContext(account)
+
+	dinosaur := h.NewDinosaur("Brontosaurus")
+
+	// We start 20 threads to update this dinosaur at the same time, for each thread, it gets the dinosaur record
+	// firstly, then updates the dinosaur species only when the species is Brontosaurus, this will cause the race
+	// condition among these threads, so we use advisory locks to avoid the race condition.
+	threads := 20
+	var wg sync.WaitGroup
+	wg.Add(threads)
+
+	updateCount := 0
+	for i := 0; i < threads; i++ {
+		go func() {
+			defer wg.Done()
+
+			if updateDinosaurWithAdvisoryLock(ctx, h.DBFactory, client, account.ID(), dinosaur.ID) {
+				updateCount = updateCount + 1
+			}
+		}()
+	}
+
+	// waits for all goroutines above to complete
+	wg.Wait()
+
+	Expect(updateCount).To(Equal(1))
+}
+
+func updateDinosaurWithAdvisoryLock(ctx context.Context, connection db.SessionFactory, client *openapi.APIClient, accountID, dinosaurID string) bool {
+	lockID := accountID + dinosaurID
+	ctx, lockOwnerID, err := db.NewAdvisoryLockContext(ctx, connection, lockID, db.RegcredLock)
```

Comment:
this lock should be used in `pkg/service/dinosaurs.go` to show its usage.  comments can be added to the function in the service to explain the lock usage. it's an example for everyone to learn from.

Reply:
@markturansky I update the replace method in the dinosaurs service as an example, please take a look, thanks